src/Common/tests/System/IO/FileCleanupTestBase.cs

```diff
@@ -78,6 +78,6 @@ protected virtual void Dispose(bool disposing)
         /// <param name="memberName">The member name of the function calling this method.</param>
         /// <param name="lineNumber">The line number of the function calling this method.</param>
         protected string GetTestFileName(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0) =>
-            string.Format(index.HasValue ? "{0}_{1}_{2}" : "{0}_{1}", memberName ?? "TestBase", lineNumber, index.GetValueOrDefault());
+            string.Format(index.HasValue ? "{0}_{1}_{2}_{3}" : "{0}_{1}_{3}", memberName ?? "TestBase", lineNumber, index.GetValueOrDefault(), Guid.NewGuid().ToString("N"));
```

Comment:
Guid.NewGuid.ToString("N") is 20 chars longer than Path.GetRandomFileName()... that really matters?  The problem with GetRandomFileName is that more tests will likely need to be changed and in more intricate ways, e.g. the changes I made here https://github.com/dotnet/corefx/pull/19482/files#diff-b25d6041f534cd64f9f1aff2401858acR391 would have to be more intense as I can assume Guid.NewGuid().ToString("N") won't contain letters other than A-F, but I can't assume that for GetRandomFileName.